src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -391,42 +391,79 @@ public Set<DatasetDTO> describeDatasets(Integer userId) {
         List<Integer> datasetIdsInUse = dataAccessRequestDAO.findAllDARDatasetRelationDatasetIds();
         HashSet<DatasetDTO> datasets = new HashSet<>();
         if (userHasRole(UserRoles.ADMIN.getRoleName(), userId)) {
-            datasets.addAll(datasetDAO.findAllDatasets());
+            datasets.addAll(datasetDAO.findAllDatasetDTOs());
         } else {
-            datasets.addAll(datasetDAO.findActiveDatasets());
+            datasets.addAll(datasetDAO.findActiveDatasetDTOs());
             if (userHasRole(UserRoles.CHAIRPERSON.getRoleName(), userId)) {
-                Collection<DatasetDTO> chairSpecificDatasets = datasetDAO.findDatasetsByUserId(userId);
+                Collection<DatasetDTO> chairSpecificDatasets = datasetDAO.findDatasetDTOsByUserId(userId);
                 datasets.addAll(chairSpecificDatasets);
             }
         }
         datasets.forEach(d -> d.setDeletable(!datasetIdsInUse.contains(d.getDataSetId())));
         return datasets;
     }
 
+
+    public List<Dataset> searchDatasets(String query, User user) {
+        List<Dataset> datasets = this.findAllDatasetsByUser(user);
+
+        return datasets.stream().filter(ds ->this.filterDatasetByQuery(ds, query)).toList();
+    }
+
+
+    // dataset properties which can be searched by
+    private static final List<String> filterableDatasetProperties = List.of(
+            "Principal Investigator(PI)"
+    );
```

Comment:
should we add more properties which can be filtered/searched upon?

Reply:
should we make it searchable on `dataUse` fields? for example, 'irb' becomes a searchable term if `ethicsApprovalRequired == true`?

Reply:
I'm a fan of searching on anything in the dataset. This is an area where we have not put much effort into because the Data Catalog team is doing something much more comprehensive and repeating some of that work in DUOS is not a priority. However, since we're here, I do think we can expand our search fields to be whatever makes the most sense.